config_reader_lib/main.cpp

```diff
@@ -0,0 +1,75 @@
+
+#include "./Assignment3-Libraries.h"
+
+namespace configio{
+
+  // Function returns a pointer to a ConfigFile struct
+  ConfigFile* read_config(string file_path){
+
+    //get number of lines in file
+    // number of lines divided by two because file input will be distributed into two parallel arrays
+    //int num_lines = line_count(file_path);
+    const int SIZE = 5;
+
+    // declare input file and open path
+    ifstream file(file_path);
+
+    // declare parallel arrays for file input
+    string headers[SIZE];
+    string values[SIZE];
+
+    //read lines of file and store in array
+    int i = 0;
+    do{
+      getline(file, headers[i]);
+      getline(file, values[i]);
+      cout << headers[i] << endl; //test for lines being read correctly from file
+      cout << values[i] << endl;
+      i++;
+    } while(!file.eof());
+
+    // file closes after finished reading input
+    file.close();
+
+    // Create string pointer variable and assign address of array
+    ConfigFile config;
+
+    // Still needs error checking to make sure fields are in correct order in file
+    config.first_name = values[0];
+    config.email = values[1];
+    config.cypher = values[2];
+    config.timezone = values[3];
+    config.filepath = values[4];
+
+    ConfigFile* configptr = &config;
```

Comment:
I would put the "new" keyword in here to make this dynamic. This information might be lost after the function ends.